yammbs/torsion/analysis.py

```diff
@@ -27,58 +33,164 @@ def _normalize(qm: dict[float, float], mm: dict[float, float]) -> tuple[dict[flo
 
-        return pandas.DataFrame(
-            [een.een for een in self],
-            index=pandas.Index([een.id for een in self]),
-            columns=["een"],
+    @classmethod
+    def from_data(
+        cls,
+        torsion_id: int,
+        molecule: "Molecule",
+        qm_points: dict[float, Array],
+        mm_points: dict[float, Array],
+    ) -> Self:
+        """Create an RMSD object by calculating the RMSD between QM and MM points."""
+        rmsd_vals = numpy.array([get_rmsd(molecule, qm_points[key], mm_points[key]) for key in qm_points])
+        return cls(
+            id=torsion_id,
+            rmsd=numpy.sqrt((rmsd_vals**2).mean()),
         )
 
-    def to_csv(self, path: str):
-        """Write this collection to a CSV file."""
-        self.to_dataframe().to_csv(path)
+
+class RMSDCollection(AnalysisMetricCollection[RMSD]):
+    """A collection of RMSD models."""
+
+    item_type = RMSD
+
+
+class RMSE(AnalysisMetric):
+    """A model storing the RMSE error over a torsion profile."""
+
+    id: int
+    rmse: float
+
+    @classmethod
+    def from_data(cls, torsion_id: int, qm_energies: Array, mm_energies: Array) -> Self:
+        """Create an RMSE object by calculating the RMSE between QM and MM energies."""
+        return cls(
+            id=torsion_id,
+            rmse=numpy.sqrt(((qm_energies - mm_energies) ** 2).mean()),
+        )
+
+
+class RMSECollection(AnalysisMetricCollection[RMSE]):
+    """A collection of RMSE models."""
+
+    item_type = RMSE
+
+
+class MeanError(AnalysisMetric):
+    """A model storing the mean error over a torsion profile."""
+
+    id: int
+    mean_error: float
+
+    @classmethod
+    def from_data(cls, torsion_id: int, qm_energies: Array, mm_energies: Array) -> Self:
+        """Create a MeanError object by calculating the mean MM - QM energy."""
+        return cls(
+            id=torsion_id,
+            mean_error=numpy.mean(mm_energies - qm_energies),
+        )
+
+
+class MeanErrorCollection(AnalysisMetricCollection[MeanError]):
+    """A collection of MeanError models."""
+
+    item_type = MeanError
+
+
+class JSDistance(AnalysisMetric):
+    """A model storing the Jensen-Shannon distances."""
+
+    id: int
+    js_distance: float
+    js_temperature: float
+
+    @classmethod
+    def from_data(
+        cls,
+        torsion_id: int,
+        qm_energies: Array,
+        mm_energies: Array,
+        temperature: float,
+    ) -> Self:
+        """Create a JSDistance object from supplied data.
+
+        Do this by calculating the Jensen-Shannon distance
+        between the two distributions generated with Boltzmann inversion at the
+        specified temperature, using base 2 logs. A distance of 0 indicates
+        identical distributions, while a distance of 1 indicates completely
+        non-overlapping distributions.
+        """
+        from scipy.spatial.distance import jensenshannon
+
+        beta = 1.0 / (temperature * 0.0019872043)  # kcal/mol to K
```

Comment:
I wonder if we want to use units here

Reply:
Thanks for the comment. I wasn't sure how much to use explicit units given the the energies are stored with implicit units of kcal/mol, but at the very least I should probably do something like:
```Python
kb = (1 * unit.avogadro_constant * unit.boltzmann_constant).to(unit.kilocalorie / unit.mole / unit.kelvin).magnitude
beta = 1.0 / (temperature * kb) # Implicit units of 1/(kcal / mol)
```
Or maybe I should also include units on the temperature argument? Thanks.